node-tests/acceptance/build-test.js

```diff
@@ -0,0 +1,41 @@
+var path = require('path');
+var chai = require('chai');
+var chaiFiles = require('chai-files');
+var compareBuild = require('../helpers/compare-build');
+
+chai.use(chaiFiles);
+
+var expect = chai.expect;
+var file = chaiFiles.file;
+
+describe('Acceptance', function() {
+  describe('build', function() {
+    this.timeout(300000);
+
+    var build;
+
+    beforeEach(function() {
+      build = compareBuild();
+    });
+
+    afterEach(function() {
+      build.cleanup();
+    });
+
+    it('correctly builds engine-testing application', function() {
+      var oldBuildFiles = build.oldFiles;
+      var newBuildFiles = build.newFiles;
+
+      // Verify the before and after builds have the same files
+      expect(newBuildFiles).to.deep.equal(oldBuildFiles);
+
+      // Expect the contents of all the build files to be the same
+      for (var i = 0; i < oldBuildFiles.length; i++) {
+        var oldFilePath = path.join(build.oldFilesPath, oldBuildFiles[i]);
+        var newFilePath = path.join(build.newFilesPath, newBuildFiles[i]);
+
+        expect(file(newFilePath)).to.equal(file(oldFilePath));
```

Comment:
`git diff`, yes; build artifacts, no. You have to upload artifacts somewhere external to Travis and I don't feel that hassle is worth it.